Open Judge System/Services/OJS.Services/Business/Contests/ContestsBusinessService.cs

```diff
@@ -194,12 +194,15 @@ public int GetContestSubmissionsAverageRunTimeSeconds(Contest contest)
                                             StartedExecutionOn = s.StartedExecutionOn,
                                             ModifiedOn = s.ModifiedOn,
                                             CreatedOn = s.CreatedOn,
+                                            CompletedExecutionOn = s.CompletedExecutionOn,
                                         }))
                                     .ToList();
 
             if (contestSubmissions.Any(s => s.StartedExecutionOn.HasValue))
             {
-                return (int)contestSubmissions.Where(s => s.StartedExecutionOn.HasValue).Average(s => (s.ModifiedOn.Value - s.StartedExecutionOn.Value).TotalSeconds);
+                return contestSubmissions.Any(s => s.CompletedExecutionOn.HasValue)
+                    ? (int)contestSubmissions.Where(s => s.StartedExecutionOn.HasValue).Average(s => (s.CompletedExecutionOn.Value - s.StartedExecutionOn.Value).TotalSeconds)
+                    : (int)contestSubmissions.Where(s => s.StartedExecutionOn.HasValue).Average(s => (s.ModifiedOn.Value - s.StartedExecutionOn.Value).TotalSeconds);
```

Comment:
This does not seem correct.
It can throw exception in `s.CompletedExecutionOn.Value` as you do not check if this has value.
`contestSubmissions.Any(s => s.CompletedExecutionOn.HasValue)` will return true if only 1 submission satisfies it, but the next call will iterate all submissions.
I think you will not need any `.Any` or `Where` caluses, just handle all possible cases when getting the average
```
contestSubmissions.Average(s =>
  ((s.StartedExecutionOn.HasValue ? s.StartedExecutionOn : s.CreatedOn)
    - (s.CompletedExecutionOn.HasValue ? s.CompletedExecutionOn : s.ModifiedOn.Value))
  .TotalSeconds)
```

Reply:
Done